pkgs/_macro_runner/lib/macro_runner.dart

```diff
@@ -2,9 +2,19 @@
 // for details. All rights reserved. Use of this source code is governed by a
 // BSD-style license that can be found in the LICENSE file.
 
+import 'dart:convert';
+import 'dart:io';
+
 import 'package:_macro_builder/macro_builder.dart';
 import 'package:macro_service/macro_service.dart';
 
+/// Runs macros.
+///
+/// TODO(davidmorgan): support shutdown/cleanup.
 class MacroRunner {
-  void run(BuiltMacroBundle macro, HostEndpoint endpoint) {}
+  /// Runs [macroBundle] connected to [endpoint].
+  void run(
```

Comment:
It seems a bit odd to not have this return a Future - maybe just call it `start`? It also seems like it should just be a top level method.